lib/autoupdate/agent/installer.go

```diff
@@ -44,11 +45,33 @@ const (
 	checksumHexLen = sha256.Size * 2 // bytes to hex
 )
 
+var (
+	// tgzExtractPaths describes how to extract the Teleport tgz.
+	// See utils.Extract for more details on how this list is parsed.
+	// Paths must use tarball-style / separators (not filepath).
+	tgzExtractPaths = []utils.ExtractPath{
+		{Src: "teleport/examples/systemd/teleport.service", Dst: "etc/systemd/teleport.service"},
+		{Src: "teleport/examples", Skip: true},
+		{Src: "teleport/install", Skip: true},
+		{Src: "teleport/README.md", Dst: "share/README.md"},
+		{Src: "teleport/CHANGELOG.md", Dst: "share/CHANGELOG.md"},
+		{Src: "teleport/VERSION", Dst: "share/VERSION"},
+		{Src: "teleport", Dst: "bin"},
+	}
+
+	// servicePath contains the path to the Teleport SystemD service within the version directory.
+	servicePath = filepath.Join("etc", "systemd", "teleport.service")
```

Comment:
should it be `/etc/systemd/system/` path? since we have such help message https://github.com/gravitational/teleport/commit/c4316308976ed2c4ba30c4ee47710e71b8708607#diff-39c53543c507ce0bdbd34343ad186bc7d95006d8a62171be800239fb556a8522R66-R78

Reply:
This is the path within the `versions` directory (e.g., `/var/lib/teleport/versions/1.2.3/etc/systemd/teleport.service`), which moves the configuration out of the `examples` directory in the Teleport tgz (`examples/systemd/teleport.service`). It's not related to the system `/etc/systemd/system/` directory, which is for custom service files.

Not strongly opinionated on if we introduce `/system/`, since only the symlink location matters to systemd. Happy to change it if anyone feel strongly. It might make more sense as `/var/lib/teleport/versions/1.2.3/lib/systemd/system/teleport.service`, to better match the symlink location.

Reply:
Merging as-is for now, since theoretically we could add options for linking to other places (e.g., `/usr/local/systemd/user/`, `/etc/systemd/system`) in the future. Also worried that we may have to change the linking destination if systemd isn't compiled with `/usr/local` paths in a supported distro (CoreOS does not support it).

Happy to change in a follow-up PR if there are any concerns.